### ISSUE TITLE
"etc/no-assign-mutated-array" not catching all occurrences

### ISSUE THREAD

**Issue by mwsiddeeq15** [cartant/eslint-plugin-etc#37]
Not sure exact why but `etc/no-assign-mutated-array` is not catching all occurrences of assigning the result of an array mutation method. It found one but I did a search `=\s\w+\.sort\(` around the app and found many occurrences it missed. The one occurrence it actually caught was similar to the ones it missed, I don't see a difference other than the fact it didn't catch them.

example of some occurrences it didn't catch:
```
someLetVariable = someLetVariable.sort((b, a) => a - b)

const sortedData = dataFromProps.sort((b, a) => a - b)

const someSortedData = this.props.data.sort((b, a) => a - b)
```

but other "etc" rules catch issues in the same file so I am sure the file itself is being linted but this rule doesnt seem to catch things.

**Comment by cartant** [comment]
There's not enough information here for this to be actionable. It's possible that the variables/props aren't typed as arrays - the rule checks the type. A snippet that demonstrates the issue would be needed.
